service/matching/matching_engine.go

```diff
@@ -703,6 +706,32 @@ func (e *matchingEngineImpl) getHistoryForQueryTask(
 	return hist, resp.GetResponse().GetNextPageToken(), err
 }
 
+func (e *matchingEngineImpl) nonRetryableErrorsDropTask(task *internalTask, taskQueueName string, namespaceID string, err error) {
+	e.logger.Info(err.Error(),
+		tag.WorkflowNamespaceID(task.event.Data.GetNamespaceId()),
+		tag.WorkflowID(task.event.Data.GetWorkflowId()),
+		tag.WorkflowRunID(task.event.Data.GetRunId()),
+		tag.WorkflowTaskQueueName(taskQueueName),
+		tag.TaskID(task.event.GetTaskId()),
+		tag.WorkflowEventID(task.event.Data.GetScheduledEventId()),
+		tag.Error(err),
+	)
+
+	// log down the metrics
+	if common.IsInternalError(err) {
+		metrics.TaskInternalErrorCounter.With(e.metricsHandler).Record(1, metrics.NamespaceTag(namespaceID))
```

Comment:
NamespaceTag should take namespace name, not id. The log line should probably use namespace name too